softwareconfiguration/mlplan/src/main/java/de/upb/crc901/mlplan/multiclass/wekamlplan/sklearn/SKLearnClassifierFactory.java

```diff
@@ -0,0 +1,139 @@
+package de.upb.crc901.mlplan.multiclass.wekamlplan.sklearn;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.upb.crc901.mlplan.multiclass.wekamlplan.ClassifierFactory;
+import hasco.exceptions.ComponentInstantiationFailedException;
+import hasco.model.CategoricalParameterDomain;
+import hasco.model.ComponentInstance;
+import hasco.model.Parameter;
+import jaicore.basic.ILoggingCustomizable;
+import jaicore.ml.scikitwrapper.ScikitLearnWrapper;
+import weka.classifiers.Classifier;
+
+/**
+ * The SKLearnClassifierFactory takes a ground component instance and parses it into a <code>ScikitLearnWrapper</code> as defined in the project jaicore-ml.
+ * This factory may be used in the context of HASCO, especially for ML-Plan.
+ *
+ * @author wever
+ */
+public class SKLearnClassifierFactory implements ClassifierFactory, ILoggingCustomizable {
+
+	private static final CategoricalParameterDomain BOOL_DOMAIN = new CategoricalParameterDomain(Arrays.asList("True", "False"));
+
+	private Logger logger = LoggerFactory.getLogger(SKLearnClassifierFactory.class);
+	private String loggerName;
+
+	@Override
+	public Classifier getComponentInstantiation(final ComponentInstance groundComponent) throws ComponentInstantiationFailedException {
+		this.logger.info("Parse ground component instance {} to ScikitLearnWrapper object.", groundComponent);
+
+		StringBuilder constructInstruction = new StringBuilder();
+		Set<String> importSet = new HashSet<>();
+		constructInstruction.append(this.extractSKLearnConstructInstruction(groundComponent, importSet));
+		StringBuilder imports = new StringBuilder();
+		importSet.forEach(imports::append);
+
+		try {
+			return new ScikitLearnWrapper(constructInstruction.toString(), imports.toString());
+		} catch (IOException e) {
+			this.logger.error("Could not create sklearn wrapper for construction {} and imports {}.", constructInstruction, imports);
+			return null;
+		}
+	}
+
+	public String extractSKLearnConstructInstruction(final ComponentInstance groundComponent, final Set<String> importSet) {

```

Comment:
is this really public?

Reply:
Otherwise it cannot be tested, but maybe this should go into a Util class?